### ISSUE TITLE
Move to a PoW engine for all tests involving an Ethereum node

### ISSUE THREAD

**Issue by thomaseizinger** [comit-network/comit-rs#1106]
Currently we are using the `instantSeal` engine of parity whenever we need an Ethereum node to test against.

However, that is quite brittle as we perceived while developing the `EthereumWallet` for bobtimus. In addition, this causes a divergence in how we structure our e2e tests, especially the `refund` tests where we need to account for time passing.

Moving to a PoW-based engine would allow us to continuously generate blocks which is hopefully more resilient.

Labelling this with `bug` because it may be involved in our build to be flacky at the moment.

Tasks:
- Use a PoW-engine for the Ethereum node in the e2e tests
- Convert the e2e tests to "wait until" the appropriate time is reached, same as we currently do for Bitcoin
- Use a PoW-engine for the Ethereum node in the `blockchain_contracts` tests

**Comment by thomaseizinger** [comment]
To be re-discussed once we have more data based on: #1119.

**Comment by bonomat** [comment]
Note: when doing this, also fix some ugly sleeps in api_tests/lib/ethereum.ts

**Comment by bonomat** [comment]
Note: this ticket was labeled as a *bug* because btsieve was missing blocks due to the instant seal of parity's `dev` setting. Since btsieve can't go into the past, this was indeed an issue. 
However, with the sleep introduced some time ago, this is not so criticial anymore. 
I've removed the *bug* label for now. 

Here some of my findings: 

PoW is not suitable for our needs due its nature of being *proof of work*, i.e. it computes a cryptographic puzzle for each block. The only way to generate blocks every X seconds is by configuring the difficulty, however, this means that the CPU will be heavily used for these X seconds. If we want to make the puzzle simpler, we would just find blocks faster. 

--> PoA is the only alternative. 

I tried to setup a parity node with a PoA consensus which generated a block every 1 second. However, I had the negative side effect that some of our tests timed out due to the lack of `waits` for transactions to being mined. 

I'm about to fix this but before finishing this properly, we will need to release a docker container for a custom PoA network. 
Do we want to go this path or do we want to stick with parity's `dev` chain?
